productcomments.php

```diff
@@ -379,53 +419,86 @@ public function renderModerateLists()
 
         $fields_list = $this->getStandardFieldList();
 
-        if (version_compare(_PS_VERSION_, '1.6', '<')) {
-            $return .= '<h1>'.$this->l('Reported Reviews').'</h1>';
-            $actions = array('enable', 'delete');
-        } else {
-            $actions = array('delete', 'noabuse');
-        }
+        $actions = array('delete', 'noabuse');
 
         $helper = new HelperList();
+        $helper->list_id = 'form-productcomments-reported-list';
         $helper->shopLinkType = '';
         $helper->simple_header = true;
         $helper->actions = $actions;
         $helper->show_toolbar = false;
         $helper->module = $this;
         $helper->listTotal = count($comments);
         $helper->identifier = 'id_product_comment';
-        $helper->title = $this->l('Reported Reviews');
+        $helper->title = $this->trans('Reported Reviews', [], 'Modules.Productcomments.Admin');
         $helper->table = $this->name;
         $helper->token = Tools::getAdminTokenLite('AdminModules');
-        $helper->currentIndex = AdminController::$currentIndex.'&configure='.$this->name;
-        //$helper->tpl_vars = array('priority' => array($this->l('High'), $this->l('Medium'), $this->l('Low')));
+        $helper->currentIndex = AdminController::$currentIndex . '&configure=' . $this->name;
 
         $return .= $helper->generateList($comments, $fields_list);
 
         return $return;
     }
 
+    /**
+     * Method used by the HelperList to render the approve link
+     *
+     * @param $token
+     * @param $id
+     * @param null $name
+     *
+     * @return mixed
+     */
+    public function displayApproveLink($token, $id, $name = null)
+    {
+        $this->smarty->assign(array(
+            'href' => $this->context->link->getAdminLink('AdminModules') . '&configure=' . $this->name . '&module_name=' . $this->name . '&approveComment=' . $id,
```

Comment:
You're the one who added a feature to this method in order to generate urls with params. You may want to use it, don't you?

Reply:
You're right. Shame on me 🥴
Done